### ISSUE TITLE
RFC: Add a CI job to run checks automatically?

### ISSUE THREAD

**Issue by cofyc** [pingcap/tidb-operator#391]
Why:

`make check` fails currently, if there is a CI job to run `make check` or `make check-all` automatically and we can know and fix it before PR is merged.

```
$ make check
# govet doesn't work with retool for Go 1.11
# so install separately
linting
CGO_ENABLED=0 retool do revive -formatter friendly -config revive.toml $($(go list ./... | grep -vE "pkg/client" | grep -vE "zz_generated"))
/bin/sh: 1: github.com/pingcap/tidb-operator/cmd/controller-manager: not found
gofmt checking
gofmt -s -l -w $(find $(go list ./... | grep -vE "pkg/client" | grep -vE "zz_generated" | sed 's|github.com/pingcap/tidb-operator/||') -name "*.go") 2>&1| awk '{ print } END { if (NR > 0) { exit 1 } }'
tests/fault.go
Makefile:88: recipe for target 'check-static' failed
make: *** [check-static] Error 1
```

How:

- travis / circle CI job
- jenkins job

**Comment by cofyc** [comment]
Oh, current Jenkins job `ci/pingcap_tidb_operator_build_dind.groovy` runs `make check`. My mistake.

**Comment by cofyc** [comment]
`make check` will be fixed in #390.
